Calendar/Check-SharingStatus.ps1

```diff
@@ -0,0 +1,326 @@
+
+# Define the parameters
+[CmdletBinding()]
+param(
+    [Parameter(Mandatory=$true)]
+    [string]$Owner,
+    [Parameter(Mandatory=$true)]
+    [string]$Receiver
+)
+
+$BuildVersion = ""
+
+. $PSScriptRoot\..\Shared\ScriptUpdateFunctions\Test-ScriptVersion.ps1
+
+if (Test-ScriptVersion -AutoUpdate) {
+    # Update was downloaded, so stop here.
+    Write-Host "Script was updated. Please rerun the command."  -ForegroundColor Yellow
+    return
+}
+
+Write-Verbose "Script Versions: $BuildVersion"
+
+$script:PIIAccess = $true #Assume we have PII access until we find out otherwise
+
+<#
+.SYNOPSIS
+    Formats the CalendarSharingInvite logs from Export-MailboxDiagnosticLogs for a given identity.
+.DESCRIPTION
+    This function processes calendar sharing accept logs for a given identity and outputs the most recent update for each recipient.
+.PARAMETER Identity
+    The SMTP Address for which to process calendar sharing accept logs.
+#>
+function ProcessCalendarSharingInviteLogs {
+    param (
+        [string]$Identity
+    )
+
+    # Define the header row
+    $header = "Timestamp", "Mailbox", "Entry MailboxOwner", "Recipient", "RecipientType", "SharingType", "DetailLevel"
+    $csvString = @();
+    $csvString = $header -join ","
+    $csvString += "`n"
+
+    Write-Output "------------------------"
+    Write-Output "Looking for modern calendar sharing accept data for [$Identity]."
+
+    # Call the Export-MailboxDiagnosticLogs cmdlet and store the output in a variable
+    try {
+        # Call the Export-MailboxDiagnosticLogs cmdlet and store the output in a variable
+        # -ErrorAction is not supported on Export-MailboxDiagnosticLogs
+        # $logOutput = Export-MailboxDiagnosticLogs $Identity -ComponentName CalendarSharingInvite -ErrorAction SilentlyContinue
+
+        $logOutput = Export-MailboxDiagnosticLogs $Identity -ComponentName CalendarSharingInvite
+    } catch {
+        # Code to run if an error occurs
+        Write-Error "An error occurred: $_"
+    }
+
+    # check if the output is empty
+    if ($null -eq $logOutput.MailboxLog) {
+        Write-Output "No data found for [$Identity]."
+        return
+    }
+
+    $logLines =@();
+    # Split the output into an array of lines
+    $logLines = $logOutput.MailboxLog -split "`r`n"
+
+    # Loop through each line of the output
+    foreach ($line in $logLines) {
+        if ($line -like "*RecipientType*") {
+            $csvString += $line + "`n"
+        }
+    }
+
+    # Clean up output
+    $csvString = $csvString.Replace("Mailbox: ", "")
+    $csvString = $csvString.Replace("Entry MailboxOwner:", "")
+    $csvString = $csvString.Replace("Recipient:", "")
+    $csvString = $csvString.Replace("RecipientType:", "")
+    $csvString = $csvString.Replace("Handler=", "")
+    $csvString = $csvString.Replace("ms-exchange-", "")
+    $csvString = $csvString.Replace("DetailLevel=", "")
+
+    # Convert the CSV string to an object
+    $csvObject = $csvString | ConvertFrom-Csv
```

Comment:
Instead of doing a csv string array, can't we just create a new PSCustomObject and add that to a list?

Reply:
After looking at the logs that was provided, I still think the object type would be better way to go about reading the data. However, I will keep this up to you. If you want to keep it as is, that is fine.